### ISSUE TITLE
For submission on KITTI 3D OD benchmark

### ISSUE THREAD

**Issue by mingyuShin** [abhi1kumar/DEVIANT#6]
Hello. Thank you for the great research. 

I have two questions.

1. When submitting to the KITTI dataset benchmark, is it correct to use all 7,481 images in the training folder for training? 

2. If so, could you please explain how you validated for submission to the test? 

Thank you.

**Comment by abhi1kumar** [comment]
Hi @mingyuShin ,
Thank you for showing interest in our work. 
 
>  1. When submitting to the KITTI dataset benchmark, is it correct to use all 7,481 images in the training folder for training?

Yes. We follow previous works and use all 7481 images for training and 7,518 images for testing. Please see Data Splits in section 5 of our [paper](https://arxiv.org/pdf/2207.10758.pdf).
 
>  2. If so, could you please explain how you validated for submission to the test?

The KITTI Test ground truth labels are hidden from the public. We submitted our KITTI Test predictions to the KITTI server. The server runs evaluation with the hidden test ground truth labels and reports the results [here](https://www.cvlibs.net/datasets/kitti/eval_object_detail.php?&result=667048f2a0d95fe11257db441ed2f27c14047ad6). We report these server-evaluated results in our paper and README of this repo. 

The complete KITTI leaderboard is [here](https://www.cvlibs.net/datasets/kitti/eval_object.php?obj_benchmark=3d).

**Comment by mingyuShin** [comment]
Thank you for your replying. @abhi1kumar 

Then, when you train for KITTI Test split, is it correct to use the seed and other hyperparameters that were found on KITTI Val split?

**Comment by abhi1kumar** [comment]
> Then, when you train for KITTI Test split, is it correct to use the seed and other hyperparameters that were found on KITTI Val split?

Yes. You should choose your hyperparameters based on the KITTI Val Split results. Then, you should train a model with these hyperparameters on all 7481 images and submit to the KITTI server to get the results.

**Comment by mingyuShin** [comment]
Thanks a lot for your replying!

**Comment by JunjieWang0528** [comment]
> Thanks a lot for your replying!

May I ask how to submit the results to the Kitt official website? I encountered an error while submitting
![image](https://github.com/abhi1kumar/DEVIANT/assets/51773333/d37e10cf-3a33-4378-af32-7d14b5a0c54f)

**Comment by JunjieWang0528** [comment]
![image](https://github.com/abhi1kumar/DEVIANT/assets/51773333/473db32a-1689-4082-95c7-1f669fdb218c)
When I submitted, the data was in the data folder

**Comment by abhi1kumar** [comment]
@1800710138 
You should **NOT** keep predictions inside the `data` folder. The submission zip should contain **only** `.txt` files and **NO** folder.

**Comment by JunjieWang0528** [comment]
Thank you, it's resolved




------------------&nbsp;原始邮件&nbsp;------------------
发件人: " Abhinav Kumar (अभिनव ***@***.***&gt;; 
发送时间: 2023年5月13日(星期六) 中午1:12
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [abhi1kumar/DEVIANT] For submission on KITTI 3D OD benchmark (Issue #6)





 
@1800710138
 You should NOT keep predictions inside the data folder. The submission zip should contain only .txt files and NO folder.

**Comment by Aangss** [comment]
![image](https://github.com/user-attachments/assets/1530d425-6c61-46af-918a-9085c9d93f47)
Must the 3 dimensions of the 3D attribute be modified to 0 for the test set results?
like this?
Car 0.0 0 -1.59 726.63 169.44 769.13 205.21 1.56 1.62 3.51 6.24 1.43 33.01 -1.40 0.14
                                                              |
                                                            ⬇
Car 0.0 0 -1.59 726.63 169.44 769.13 205.21 0 0 0 6.24 1.43 33.01 -1.40 0.14
Or is it better to just compress the results without any changes?

**Comment by abhi1kumar** [comment]
> Must the 3 dimensions of the 3D attribute be modified to 0 for the test set results?
like this?

You should **NOT** do any modification. As an example, you could take our results from the KITTI test model and submit on the KITTI website.

**Comment by Aangss** [comment]
Thank you

**Comment by Aangss** [comment]
@abhi1kumar 
Excuse me, I have a question, when using trainval training, how can I know which epoch has the best effect, or use val as the evaluation standard?

**Comment by abhi1kumar** [comment]
> how can I know which epoch has the best effect

We do not know. We report the results at the end of training (which is 140th epoch checkpoint) for all val and (test) leaderboard models. Another reason of not doing checkpoint selection is it makes the results irreproducible.

**Comment by Aangss** [comment]
> > how can I know which epoch has the best effect我怎么知道哪个epoch效果最好
> 
> We do not know. We report the results at the end of training (which is 140th epoch checkpoint) for all val and (test) leaderboard models. Another reason of not doing checkpoint selection is it makes the results irreproducible.我们不知道。我们在训练结束时（即第 140 个纪元检查点）报告所有验证和（测试）排行榜模型的结果。不进行检查点选择的另一个原因是它会使结果不可重现。

Thanks
